hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMSecurityProtocolServer.java

```diff
@@ -403,13 +415,20 @@ public List<String> listCACertificate() throws IOException {
   }
 
   @Override
-  public String getRootCACertificate() throws IOException {
-    LOGGER.debug("Getting Root CA certificate.");
-    if (storageContainerManager.getScmStorageConfig()
-        .checkPrimarySCMIdInitialized()) {
-      return CertificateCodec.getPEMEncodedString(rootCACertificate);
+  public synchronized String getRootCACertificate() throws IOException {
+    Date lastCertDate = new Date(0);
+    X509Certificate lastCert = null;
+    for (X509Certificate cert : rootCACertificate) {
+      if (cert.getNotAfter().after(lastCertDate)) {
+        lastCertDate = cert.getNotAfter();
+        lastCert = cert;
+      }
     }
-    return null;
+    return CertificateCodec.getPEMEncodedString(lastCert);
```

Comment:
Before calling need to ensure `lastCert` is not NULL or else it will lead to NPE in `CertificateCodec .getPEMEncodedString`

Reply:
Thank you, that was a nice catch, I've added functionality to handle that